src/tests/gssapi/t_ccselect.py

```diff
@@ -92,31 +106,41 @@
 if output != (zaphod + '\n'):
     fail('zaphod not chosen as default initiator name for server in r1')
 
-# Check that primary cache is used if server realm is unknown.
-output = r2.run(['./t_ccselect', gssserver])
+# Check ccache selection using a fallback realm.
+output = r1.run(['./t_ccselect', gssserver])
+if output != (alice + '\n'):
+    fail('alice not chosen via primary cache for unknown server realm')
```

Comment:
My understanding of what happens here doesn't match the failure message.  alice will be chosen via the realm module because of the fallback realm, not via the primary ccache.

The same goes for the later tests.  I don't think you have a remaining test where the realm module produces no result, so we no longer have coverage of the case where krb5_cc_select() returns no result when invoked by kg_cred_resolve().